enhancements/declarative-index-config.md

```diff
@@ -0,0 +1,483 @@
+Currently, a bundle can be added into the index using 
+`opm index add --bundles <list-of-bundle-paths> --tag=<index-image-tag>` 
+where the bundle images (like quay.io/random/etcdv0.0.1) are included in `<list-of-bundle-path>`. The `opm index add` command will :
+
+1. Be enhanced to parse the bundle images that can include a `@channel` delimiter, followed by a channel name. 
+So for example if a user wants to add the bundle `quay.io/random/etcdv0.0.1` to the channel `alpha`, the user can express his/her intent to do so in the following manner: 
+
+```bash=
+$ opm index add --bundles quay.io/random/etcdv0.0.1@channel:alpha --tag=docker.io/mynamespace/community-operators
+```
+2. Be enhanced to create a json for representing the package the bundle is being added to if it is the first bundle in the package that is being added to this index, or add the bundle metadata to the existing package json representation if previous bundles were added that belong to the same package.
+
+#### Using the package json representation
+
+##### Introducing a new `represent` subcommand for `opm index`
+
+A new subcommand `represent` will be introduced under `opm index` which will have a subcommand of its own called `packages`.
+
+When `opm index represent packages` is summoned, the json representations of the packages inside the index will be unfurled in a folder with the same name as the index. 
+When only `opm index represent` is summoned for an index, the json representations of the packages will be aggregated to output a json representation of the entire index. 
+
+##### Introducing a new `update` subcommand for `opm index`
+
+A new subcommand `update` will be introduced under `opm index`, which will take a folder containing json representations of packages as it's input along with an existing index image tag, and will push any update made to the representations up to the index container image. 
+
+> Note: For a seamless UX, the following things should be consider during implementation when the `opm index update` command is summoned: 
+> 1. Validate the files inside the folder given as input to the command to ensure they are valid json representation of indexes (and not random files with random content, which could be potential security risks)
+> 2. If the files are valid files, validate that the index image tag given as input to the command is a valid tag, i.e and image for that tag already exists. If not, throw an error. 
+> 3. If the files are valid, and the index image tag exists, take a gitdiff between the present content of the index (i.e the existing json representations) and the content of the folder given as input. Display the diff as an output of the command with the following message 
+> "The following changes will be updated in the index:"
+> If there is no diff between the content in the container image and the content of the local folder, throw an error with the message "No diff found between index and local content to update."
```

Comment:
how are merge conflicts handled?

Reply:
@bparees could you explain a bit more on what you mean by merge conflict. Do you still have this question on the recent version of this enhancement?

Reply:
i think i understand the flow now.